### PR TITLE
Hook more properties for transient units

### PR DESCRIPTION
**Pull request by alkino**
systemd-run can now launch units with EnvironmentFile set.

From the last PR I add "empty_array" to clear current files list. And the same with an empty path in array.

### CODE REVIEW

File: src/core/dbus-execute.c [hunk 1]
```diff
@@ -1140,6 +1140,61 @@ int bus_exec_context_set_transient_property(
...
+                                        buf = strjoin(b ? "-" : "", path, NULL);
+                                        if (buf == NULL)
+                                                return -ENOMEM;
+
+                                        r = strv_push(&c->environment_files, buf);
+                                        if (r < 0)
+                                                return r;
+                                }
+
+                                unit_write_drop_in_private_format(u, mode, name, "EnvironmentFile=%s\n", buf);
```
Comment (evverx) [inline src/core/dbus-execute.c]: Try:

```
# systemd-run --property EnvironmentFile=-/a --property EnvironmentFile=-/b --remain-after-exit env
Running as unit UNITNAME

# systemctl cat UNITNAME
...
# /run/systemd/system/run-9132.service.d/50-EnvironmentFiles.conf
[Service]
EnvironmentFile=-/b
...
```

Should be:

```
# /run/systemd/system/run-9132.service.d/50-EnvironmentFiles.conf
[Service]
EnvironmentFile=-/a
EnvironmentFile=-/b
```

Comment (alkino) [inline src/core/dbus-execute.c]: The problem is that I call unit_write_drop_in_private_format twice in this case and the second drop precedent file.
Is there a way to go around this problem?

Comment (evverx) [inline src/core/dbus-execute.c]: I think, you can join the strings from the vector `c->environment_files`. There is a function [`strv_join`](https://github.com/systemd/systemd/blob/0a41d91f3022ae5107151ebd1af9b65a55030c7e/src/basic/strv.c#L348).

Comment (poettering) [inline src/core/dbus-execute.c]: Well, if you have complex strings to write, consider using strv_join() to merge them, or use a memstream for them. We do that for some other properties like this.

**Pull request closed without merging**